ios/MarigoldSDKReactNativeTests/RNMessageStreamTests.m

```diff
@@ -233,6 +235,66 @@ -(void)clearMessages:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectB
             [[check should] equal:error];
         });
     });
+    context(@"the useDefaultInAppNotification method", ^{
+        it(@"should set the default value as YES", ^{
+            [rnMessageStream useDefaultInAppNotification:YES];
+            
+            [[theValue(rnMessageStream.defaultInAppNotification) should] equal:theValue(YES)];
+        });
+        it(@"should set the default value as NO", ^{
+            [rnMessageStream useDefaultInAppNotification:NO];
+            
+            [[theValue(rnMessageStream.defaultInAppNotification) should] equal:theValue(NO)];
+        });
+    });
+    context(@"the shouldPresentInAppNotificationForMessage method", ^{
+        __block MARMessage *marMessage;
+        
+        beforeEach(^{
+            rnMessageStream.eventSemaphore = dispatch_semaphore_create(0);
+
+            marMessage = [[MARMessage alloc] init];
+            marMessage.title = @"Testing";
+            marMessage.type = MARMessageTypeText;
+            marMessage.text = @"Test Body";
+            marMessage.attributes = @{@"attributeKey": @"attributeValue"};
+        });
+
+        context(@"the shouldPresentInAppNotificationForMessage method", ^{
+            it(@"should return YES when defaultInAppNotification is YES", ^{
+                [rnMessageStream useDefaultInAppNotification:YES];
+
+                dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+                    BOOL check = [rnMessageStream shouldPresentInAppNotificationForMessage:marMessage];
```

Comment:
This test shouldn't need an async dispatch and it might even be causing the test to pass without executing the check.